### PR TITLE
Notebook Cell Event Cleanup/Rationalization

### PR DESCRIPTION
**Pull request by chlafreniere**
Found when fixing #19068, and noodling around some cell eventing fixes here so looking for some early feedback, so that's why this is a draft for now.

There are 2 events that I'm focusing on here, which were previously used quite heavily:

- Changing the edit state of a cell (boolean)
- Changing the markdown edit mode of a text cell (CellEditModes)

In a few years, these events have been overloaded and we had a bit of a mess where some events were fired multiple times and/or called during operations that weren't intended.

My attempt here is to:
1. Clean up naming conventions (feedback welcome)
2. Ensure events are only fired when they should be fired
3. Do basic cleanup in related areas

Previous to this work, markdown cells were getting re-rendered 3(!!!) times when selecting + subsequently setting the cell's `isEditMode` flag to true. I expect for large cells, there will also be a nice perf benefit as well.

### CODE REVIEW

File: src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts [hunk 1]
```diff
@@ -191,27 +192,38 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 		this.previewFeaturesEnabled = this._configurationService.getValue('workbench.enablePreviewFeatures');
 		this._register(this.themeService.onDidColorThemeChange(this.updateTheme, this));
 		this.updateTheme(this.themeService.getColorTheme());
+		this.updatePreview();
 		this.setFocusAndScroll();
-		this.cellModel.isEditMode = false;
 		this._htmlMarkdownConverter = this._instantiationService.createInstance(HTMLMarkdownConverter, this.notebookUri);
-		this._register(this.cellModel.onOutputsChanged(e => {
+		this._register(this.cellModel.onOutputsChanged(_e => {
```
Comment (corivera) [inline src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts:198]: Why the underscore here?

Comment (chlafreniere) [reply src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts:198]: vscode was complaining to me that unused variables should have underscores. We don't have a linting rule for this, so it didn't cause the build to fail. Just saw it. Happy to revert, though.

Comment (corivera) [reply src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts:198]: Can't you replace the arg with empty parentheses if it's not being used?

**Review by corivera — APPROVED** [review]

**Comment by corivera** [comment]
@chlafreniere Was looking into that failing smoke test, and it looks like the markdown toolbar buttons are broken for WYSIWYG with these changes.

**Comment by chlafreniere** [comment]
> @chlafreniere Was looking into that failing smoke test, and it looks like the markdown toolbar buttons are broken for WYSIWYG with these changes.

Thanks @corivera I'll take a look.

**Comment by barbaravaldez** [comment]
@chlafreniere I tested locally and I saw the same issue as @corivera. The toolbar does not work in WYSIWYG, but it does in markdown and split view.

Also, the other thing that I noticed is that we are not resetting to the default text edit mode after changing it when editing a text cell.
For instance, my default edit mode is Rich Text and  if I enter a text cell and change it to markdown and then exit and enter the text cell again, the edit mode remains in Markdown.

**Comment by chlafreniere** [comment]
Thanks @barbaravaldez. Will get back to this when I can. Great feedback -- appreciate it!

**Pull request closed without merging**